_drafts/oryx/technology.md

```diff
@@ -0,0 +1,60 @@
+Oryx aims to use JavaScript during development, but tries avoid the need to run JavaScript in the browser. Wherever possible, HTML and CSS are preferred to render a fast loading experience cross different screen sizes.
+
+We will examine both the deployed technology stack and the technologies utilized for development in order to gain a more comprehensive understanding of the technologies.
+
+## Deployed stack:
+
+### HTML & CSS & JS
+HTML, CSS, and JS are considered to be the fundamental and essential set of web technologies used for creating web pages. HTML is responsible for the structure and content of the page, CSS is used for styling and layout, and JS adds interactivity and dynamic behavior to the page. Together, these three technologies form the foundation of modern web development.
+
+### Web Components
+Web Components are a set of web platform APIs that allow developers to create custom, reusable, and encapsulated HTML elements.
+
+### Web hosting
+Oryx uses Netlify as a hosting provider, which means you can easily find basic boilerplate to deploy your site on Netlify. However, if you choose to use a different hosting provider, there may be limitations on the implementation of SSR. To make it work, you would need to adjust the SSR rendering boilerplate, which is specific to the hosting provider.
+
+### CSR and SSR
+Oryx provides both client-side rendered (CSR) and server-side rendered (SSR) versions of websites, so you can choose the option that best fits your needs.
+
+Client-side rendering involves loading a minimal HTML document and then using JavaScript to build the full page in the browser. This approach is great for creating highly dynamic and interactive user interfaces, as it allows for fast, seamless updates without page refreshes. However, it can negatively impact initial load time and SEO, as search engines may have trouble indexing content.
+
+Server-side rendering, on the other hand, involves generating a complete HTML document on the server and then sending it to the browser. This approach provides faster initial load times, better SEO, and improved accessibility. However, it can be more complex to implement and may not be as effective for highly dynamic pages that require frequent updates.
+
+## Development stack:
+
+### HTML
+HTML is the backbone of the web. It is the markup language used to create web pages and applications. In Oryx, HTML is used to structure the content and layout of the application. It provides a semantic structure to the page, making it easier to understand and maintain.
+
+### CSS
+CSS is a style sheet language used to describe the visual presentation of a document written in HTML. In Oryx, CSS is used to style the components and layout of the application. It allows developers to create a consistent visual style across the application and ensure that it is responsive and accessible.
+
+Oryx uses standard CSS syntax and does not utilize CSS preprocessors or post-processors, such as Sass or PostCSS. 
```

Comment:
- We should also leave a not about css in js I believe. 
- Sass and PostCSS techniques are no longer used since they add little value in a modern web framework.
  - PostCSS is no longer a big thing since browsers are much more aligned
  - Sass is no longer needed since Oryx does not use a global CSS utility library; instead, we have isolated styles per component and because of that we do not want to load global css utils in each component.

Reply:
Hm, it looks like we are going deeper and deeper into `we are not using` mode. I'm kind of agree with preprocessors, as long as they are really wide used. But why we want to add anything else we are using? I think there is no added value in mentioning we are not using PostCSS. And the same about css-in-js.